### ISSUE TITLE
Unable to Load vehicle into Carla

### ISSUE THREAD

**Issue by srpn97** [tud-hri/joan#32]
I followed the documentation to install and JOAN, but I am not seeing DemoMap or the EpicAudi class in Unreal Editor. And Without adding epicAudi as the class for the new car, JOAN doesnt spawn a vehicle on start.  I can see these files in explorer, but when I try to open Level > DemoMap, I cannot see anything, the folder is empty.

**Comment by OlgerSiebinga** [comment]
Hi @srpn97, 

It sounds like the JOAN assets are not in the correct folders. The Demomap should be in the same folder as the other Carla levels. The same goes for the blueprints of the Audi. If they are in the right folder you should be able to add the Audi to the list of vehicles (step 2 in https://joan.readthedocs.io/en/develop/setup-joan/#adding-joan-assets). Did that step work for you?

**Comment by srpn97** [comment]
In that step, I am not able to find BP_EpicAudi2 in the list. 
 Right now I have files in this location carla\Unreal\CarlaUE4\Content\JOAN Assets
 I also tried putting the same assets in carla\Unreal\CarlaUE4\Content\Carla, basically merged with the existing folder structure of carla assets.

**Comment by OlgerSiebinga** [comment]
I you cannot find BP_EpicAudi2 in the list it means the files are not in the correct folder. You should copy everything inside the folder `JOAN Assets` to the folder `carla\Unreal\CarlaUE4\Content\Carla`. The internal folder structure of `JOAN Assets` should be the same as the internal structure of `Carla`, so all the subfolders should automatically be merged. (i.e., you're adding extra files to specific subfolders of the `Carla` folder. If you then restart the unreal editor you should be able to see `BP_EpicAudi2` in the list.

**Comment by srpn97** [comment]
I have another question, will not adding these effect JOAN? From JOAN HQ in carla interface settings I should be able to choose another vehicle and should be able to control the vehicle using the keyboard right? Are these two errors related? I am not able to spawn any vehicle right now through JOAN, but I was able to run manual_control.py in the example scripts with pygame.

**Comment by OlgerSiebinga** [comment]
I'm not completely sure if these are related. We do add some functionality to the player pawn with the JOAN assets. do you get an error message when you try to spawn a vehicle from JOAN?

**Comment by srpn97** [comment]
> I you cannot find BP_EpicAudi2 in the list it means the files are not in the correct folder. You should copy everything inside the folder `JOAN Assets` to the folder `carla\Unreal\CarlaUE4\Content\Carla`. The internal folder structure of `JOAN Assets` should be the same as the internal structure of `Carla`, so all the subfolders should automatically be merged. (i.e., you're adding extra files to specific subfolders of the `Carla` folder. If you then restart the unreal editor you should be able to see `BP_EpicAudi2` in the list.

I did this, but still doesn't show up. Should I rebuild Python API? Coz the folder structure wasn't right when I first built Python API, then I realized my mistake and corrected the folder structure. I am able to to see DemoMap folder in UE Editor Content Browser, but it doesn't recognize any files in it.

**Comment by srpn97** [comment]
> I'm not completely sure if these are related. We do add some functionality to the player pawn with the JOAN assets. do you get an error message when you try to spawn a vehicle from JOAN?

It doesn't show any error, does not crash, but there is no vehicle spawn and no data collection. Its just in the running state until I stop.

**Comment by OlgerSiebinga** [comment]
> > I'm not completely sure if these are related. We do add some functionality to the player pawn with the JOAN assets. do you get an error message when you try to spawn a vehicle from JOAN?
> 
> It doesn't show any error, does not crash, but there is no vehicle spawn and no data collection. Its just in the running state until I stop.

Ok, in that case I think it's better to make a separate issue for this.

**Comment by OlgerSiebinga** [comment]
> > I you cannot find BP_EpicAudi2 in the list it means the files are not in the correct folder. You should copy everything inside the folder `JOAN Assets` to the folder `carla\Unreal\CarlaUE4\Content\Carla`. The internal folder structure of `JOAN Assets` should be the same as the internal structure of `Carla`, so all the subfolders should automatically be merged. (i.e., you're adding extra files to specific subfolders of the `Carla` folder. If you then restart the unreal editor you should be able to see `BP_EpicAudi2` in the list.
> 
> I did this, but still doesn't show up. Should I rebuild Python API? Coz the folder structure wasn't right when I first built Python API, then I realized my mistake and corrected the folder structure. I am able to to see DemoMap folder in UE Editor Content Browser, but it doesn't recognize any files in it.

Yes, you could try re-running `make launch` and `make pythonapi`

**Comment by srpn97** [comment]
> > > I you cannot find BP_EpicAudi2 in the list it means the files are not in the correct folder. You should copy everything inside the folder `JOAN Assets` to the folder `carla\Unreal\CarlaUE4\Content\Carla`. The internal folder structure of `JOAN Assets` should be the same as the internal structure of `Carla`, so all the subfolders should automatically be merged. (i.e., you're adding extra files to specific subfolders of the `Carla` folder. If you then restart the unreal editor you should be able to see `BP_EpicAudi2` in the list.
> > 
> > 
> > I did this, but still doesn't show up. Should I rebuild Python API? Coz the folder structure wasn't right when I first built Python API, then I realized my mistake and corrected the folder structure. I am able to to see DemoMap folder in UE Editor Content Browser, but it doesn't recognize any files in it.
> 
> Yes, you could try re-running `make launch` and `make pythonapi`

Thank you! I’ll try the whole setup again, and give an update here. 

Thanks again for the quick response.

**Comment by srpn97** [comment]
Hello again, I redid the whole setup, I still am facing the same issue. Joan assets are not showing up in Carla Unreal Editor. Can you share the exact version these assets were created with? I am using Unreal Engine version 4.24.3, which should be okay accoridng to the documentation. All other assets seem to work except DemoMap and BP_EpicAudi2. 

Thank you!

**Comment by OlgerSiebinga** [comment]
Currently. there are two versions of JOAN, each with its own assets and dependencies. The version on the [master branch is v1.1.0](https://joan.readthedocs.io/en/latest/), the version on the [develop branch is v1.2.0](https://joan.readthedocs.io/en/develop/). This develop branch version is not really released yet because we experienced some weird behavior in initializing the sensodrive steeringwheels (sometimes you have to do that manually and sometimes not, nothing dangerous, just annoying). The corresponding version numbers are in the documentation, and the corresponding assets are available by switching the version in the 4TU data repository. 

Which version of JOAN are you using?

**Comment by FallenWyvern** [comment]
I'm actually having the same issue. I'll note if there are two branches (1.1 and 1.2), the only "JOAN ASSETS" you can download are labelled 1.2 so maybe this is a situation where the required file isn't available?

All my folders are correct as per this discussion. I've followed all the CARLA and JOAN documentation up to this point, but the asset browser and vehicle factory both do not show the epic audi car. The uasset file exists in the directory ("/blueprints/vehicles/epicaudi/bp_epicaudi2.uasset") but it is not in the editor at all. The file has a modified date of 2022-08-29 if that means anything to the JOAN team.

**Comment by FallenWyvern** [comment]
For what it's worth, my demo map folder also exists but the assets are not present.

**Comment by OlgerSiebinga** [comment]
hi @FallenWyvern, 

Which version of JOAN are you using? The reason that there are two versions of the assets is that they only work on the specific versions of the unreal engine and CARLA associated with the two versions of JOAN (CARLA 0.9.9 for JOAN 1.1 or CARLA 0.9.13 for JOAN 1.2). 

On the data repository you can switch between the assests for [JOAN 1.1](https://data.4tu.nl/datasets/9d2b2eb9-8399-44ab-ad2f-6d2a76c00e90/1) and [JOAN 1.2](https://data.4tu.nl/datasets/9d2b2eb9-8399-44ab-ad2f-6d2a76c00e90/3) using the dropdown box directly above the authors names. 

If you're not using Sensodrive steering wheels I would recommend using JOAN 1.2 (which is on the develop branch). In that case, remember to switch to the [proper documentation page](https://joan.readthedocs.io/en/develop/) as well.

**Comment by FallenWyvern** [comment]
> hi @FallenWyvern,
> 
> Which version of JOAN are you using? The reason that there are two versions of the assets is that they only work on the specific versions of the unreal engine and CARLA associated with the two versions of JOAN (CARLA 0.9.9 for JOAN 1.1 or CARLA 0.9.13 for JOAN 1.2).
> 
> On the data repository you can switch between the assests for [JOAN 1.1](https://data.4tu.nl/datasets/9d2b2eb9-8399-44ab-ad2f-6d2a76c00e90/1) and [JOAN 1.2](https://data.4tu.nl/datasets/9d2b2eb9-8399-44ab-ad2f-6d2a76c00e90/3) using the dropdown box directly above the authors names.
> 
> If you're not using Sensodrive steering wheels I would recommend using JOAN 1.2 (which is on the develop branch). In that case, remember to switch to the [proper documentation page](https://joan.readthedocs.io/en/develop/) as well.

Hey @OlgerSiebinga 
This worked, it was absolutely binary file versioning. Whenever someone runs into "this doesn't show up", I'd be willing to bet it's that. @srpn97 should see this recommendation as a solution to their problems as well.
